src/pages/AdminDashboard/CreateRoomForm.tsx

```diff
@@ -0,0 +1,265 @@
+import Divider from '@material-ui/core/Divider';
+import Grid from '@material-ui/core/Grid';
+import { makeStyles, Theme, createStyles, withStyles } from '@material-ui/core/styles';
+import Flags from 'country-flag-icons/react/3x2'
+import {parsePhoneNumber} from 'libphonenumber-js'
+
+
+
+
+const Button = styled(MuiButton)(spacing);
+
+const CreateRoomForm = () => {
+
+    interface Values {
+        roomName: string;
+        participants: Array<any>;
+        startAt : Date;
+    }
+
+    const [value, setValue] = React.useState([]);
+
+
+    const ErrorChip = withStyles({
+        root: {
+
+            borderColor: "red"
+          
+        }
+      })(Chip);
+
+    const useStyles = makeStyles((theme: Theme) =>
+        createStyles({
+            chip : {
+                borderColor : "green"
+            },
+            chipIconContainer: {
+                width: 'fit-content'
+            },
+            flagIcon:{
+                height:'0.8rem',
+                marginRight:theme.spacing(1),
+                marginLeft:theme.spacing(1)
+            }
+        }),    
+    );
+
+    const classes = useStyles();
+
+    const handleKeyDown = (event) => {
+        switch (event.keyCode) {
+          case 186: // ;
+          case 32:  // space
+          case 9:{  // tab
+            event.preventDefault();
+            event.stopPropagation();
+            if (event.target.value.length > 0) {
+                const newValue = [...value, event.target.value] as never[]
+                setValue(newValue);
+            }
+            break;
+          }
+          default:
+        }
+      };
+
+    const  validateEmail = (email) =>{
+        const regexp = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+        return regexp.test(email);
+    }
+
+    const validatePhoneNumber = (phoneNumber) =>{
+        let isValid = true;
+
+        try {
+            parsePhoneNumber(phoneNumber,'FR');
+        }catch(error){
+            isValid = false;
+        }
+
+        return isValid;
+    }
+
+    const isParticipantValid = (participant) => {
+        return validateEmail(participant) || validatePhoneNumber(participant);
+    }
+    interface MyInputProps {
+        onKeyDown: (event: object) => void;
+    }
+    interface MyParams extends AutocompleteRenderInputParams {
+        inputProps: MyInputProps;
+    }
+
+    const renderFlagIcon = (participant) => {
+        try {
+            const phoneNumber = parsePhoneNumber(participant,'FR');
+            const FlagIcon = Flags[phoneNumber.country];
+            return ( 
+            <Grid container alignItems="center" className={classes.chipIconContainer}>
+                <FlagIcon title="United States" className={classes.flagIcon}/>
+                <Divider orientation="vertical" flexItem/>
+            </Grid>
+            );
+        }catch(error){
+            return null;
```

Comment:
I don't think it is useful to display or log as it is not an error case.

In fact parsePhoneNumber of libphonenumber-js throw an exception if the phone number has a wrong format.

In this case I use this information to not display a flag (return null element) in case of e-mail for exemple.

We can interpret this catch as : "Do not return a flag element if the input text is not a phone".